src/Pos/CLI.hs

```diff
@@ -310,7 +310,7 @@ sysStartOption = Opt.option (Timestamp . sec <$> Opt.auto) $
     Opt.value   staticSysStart
 
 peerIdOption :: Opt.Parser PeerId
-peerIdOption = fmap (PeerId . BSC.pack) $ Opt.strOption $
+peerIdOption = Opt.option (fromParsec peerIdParser) $
     Opt.long    "peer-id" <>
     Opt.metavar "PEERID" <>
     Opt.help    "Identifier for this node"
```

Comment:
Please add more verbose description here (base64, 14 bytes) and around `peerIdParser` to avoid future confusion. It'd be also good to `fail` in `peerIdParser` if the length of bytestring doesn't match.